Improve stability of AbandonedCartTest [MAILPOET-4723]

costasovo opened the pull request:
## Description

This PR should fix the flakiness of the AbandonedCartTest integration test.

## Code review notes

_N/A_

## QA notes

_N/A_

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-4723]

## After-merge notes

_N/A_


[MAILPOET-4723]: https://mailpoet.atlassian.net/browse/MAILPOET-4723?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

costasovo committed:
Add helper method for comparing DateTimeInterface objects

This commit adds a method to the integration tester that allows
comparing two DateTimeInterface objects and specify tolerated delta.
It also allows passing null and assert the DateTimeInterface internally.
This is because often our entities have return type DateTimeInterface|null
So with the internal checks we don't have to make those instance of checks
each time we pass date from entity into the method.
Note: I was not able to use $this->assertInstanceOf because PHPStan was not accepting that
and still complained.
[MAILPOET-4723]

costasovo committed:
Fix flaky AbandonedCartTest

Fix the flaky DateTime comparisons test by allowing 1 second delta
[MAILPOET-4723]

lysyjan commented on mailpoet/tests/_support/IntegrationTester.php:
```diff
@@ -98,4 +98,18 @@ public function deleteTestWooOrders() {
+   * Compares two DateTimeInterface objects by comparing timestamp values.
+   * $delta parameter specifies tolerated difference
+   */
+  public function assertEqualDateTimes(DateTimeInterface $date1 = null, DateTimeInterface $date2 = null, int $delta = 0) {
```
Can I ask why you use all parameters as optional with default values?
In my opinion, the first two parameters can be nullable without a default value. Because if I use this function without `date2`, it throws an exception. Or am I missing something?

I like this idea 👍

costasovo replied:
@lysyjan Good catch! That's exactly what I wanted to do, but my hands were probably faster than my head... Updated!

lysyjan reviewed (commented):
I have a question about the new assert parameters.
Maybe there is a reason for your solution.

costasovo committed:
Improve assertEqualDateTimes helper interface

Make sure that both dates parameters have to be passed into the method.
[MAILPOET-4723]

lysyjan reviewed (approved):
👍
